pkg/service/association.go

```diff
@@ -58,22 +58,29 @@ func isDone(ctx context.Context) bool {
 }
 
 func ensureSetupPfcpAssociation(ctx context.Context, upf *smf_context.UPF, upfStr string) {
-	var alertTime time.Time
+	alertTime := time.Now()
 	alertInterval := smf_context.SMF_Self().AssociationSetupFailedAlertInterval
+	retryInterval := smf_context.SMF_Self().AssociationSetupFailedRetryInterval
 	for {
 		err := setupPfcpAssociation(upf, upfStr)
 		if err == nil {
 			return
 		}
+		logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
 		now := time.Now()
-		if alertTime.IsZero() || now.After(alertTime.Add(alertInterval)) {
-			logger.AppLog.Errorf("Failed to setup an association with UPF%s, error:%+v", upfStr, err)
+		logger.AppLog.Debugf("now %+v, alertTime %+v", now, alertTime)
+		if now.After(alertTime.Add(alertInterval)) {
+			logger.AppLog.Errorf("ALERT for UPF%s", upfStr)
```

Comment:
@YoheiKonno , I added this log line as a future placeholder to contain the logic to send the alert to some service that will consume it.

Reply:
Thanks, LGTM

Reply:
I think associationSetupFailedAlertInterval is no more needed, just print the failed ALERT when association setup failed.